rewrite-java/src/main/java/org/openrewrite/java/SimpleMethodMatcher.java

```diff
@@ -23,80 +23,90 @@
 
 import java.util.List;
 
+import static org.openrewrite.java.tree.TypeUtils.fullyQualifiedNamesAreEqual;
+
 /**
  * A simpler version of {@link MethodMatcher} that allows for custom matching logic.
+ * <p>
+ * Defines four predicates that, in the general sense, can be used to define if a method matches:
+ * <ul>
+ *     <li>{@link #isMatchOverrides()}</li>
+ *     <li>{@link #matchesTargetTypeName(String)}</li>
+ *     <li>{@link #matchesMethodName(String)}</li>
+ *     <li>{@link #matchesParameterTypes(List)}</li>
+ * </ul>
  */
-public interface SimpleMethodMatcher {
-
-    default boolean matches(@Nullable Expression maybeMethod) {
-        return (maybeMethod instanceof J.MethodInvocation && matches((J.MethodInvocation) maybeMethod)) ||
-               (maybeMethod instanceof J.NewClass && matches((J.NewClass) maybeMethod)) ||
-               (maybeMethod instanceof J.MemberReference && matches((J.MemberReference) maybeMethod));
-    }
+public interface SimpleMethodMatcher extends BasicMethodMatcher {
 
-    default boolean matches(@Nullable J.MethodInvocation method) {
-        if (method == null) {
-            return false;
-        }
-        if (method.getMethodType() == null) {
+    @Override
+    default boolean matches(@Nullable JavaType.Method type) {
+        if (type == null) {
             return false;
         }
-        if (!matchesTargetType(method.getMethodType().getDeclaringType())) {
+        if (!matchesTargetType(type.getDeclaringType())) {
             return false;
         }
 
-        if (!matchesMethodName(method.getSimpleName())) {
+        if (!matchesMethodName(type.getName())) {
             return false;
         }
 
-        return matchesParameterTypes(method.getMethodType().getParameterTypes());
+        return matchesParameterTypes(type.getParameterTypes());
     }
 
-    default boolean matches(@Nullable J.NewClass constructor) {
-        if (constructor == null) {
-            return false;
-        }
-        JavaType.FullyQualified type = TypeUtils.asFullyQualified(constructor.getType());
-        if (type == null || constructor.getConstructorType() == null) {
-            return false;
-        }
```

Comment:
We didn't need this logic, and `matches(@Nullable J.NewClass constructor)` didn't need to be different from the rest of the other methods.